clang/test/SemaOpenCL/invalid-kernel-attrs.cl

```diff
@@ -42,7 +42,7 @@ kernel __attribute__((intel_reqd_sub_group_size(-1))) void kernel16() {} // expe
 kernel __attribute__((intel_reqd_sub_group_size(8))) __attribute__((intel_reqd_sub_group_size(16))) void kernel17() {} //expected-warning{{attribute 'intel_reqd_sub_group_size' is already applied with different parameters}}
 
 __kernel __attribute__((work_group_size_hint(8,-16,32))) void neg1() {} //expected-error{{'work_group_size_hint' attribute requires a non-negative integral compile time constant expression}}
-__kernel __attribute__((reqd_work_group_size(8,16,-32))) void neg2(){} // expected-error{{'reqd_work_group_size' attribute requires a non-negative integral compile time constant expression}}
+__kernel __attribute__((reqd_work_group_size(8, 16, -32))) void neg2() {} //expected-warning{{the resulting value of the 'reqd_work_group_size' attribute parameter 2 is always non-negative after implicit conversion}}
```

Comment:
Does it make sense to use words "first/second/third" to refer particular argument? Imagine the following example: arguments are (2, 2, -4) it says "parameter 2" which can be confusing because there was actually value "2" in attribute arguments. BTW, it says "parameter 2", but it means **third** arguments. I know that some C++ programmers even count from 0, but I believe counting from 1 and using words for it is more people friendly :)
As I can see practice with inserting thing like `%select{first|second|third}0` to a diagnostic already exists, I think we can give it a try here.

Reply:
Yes, it makes sense to me. Something like this:
the resulting value of the 'reqd_work_group_size' attribute parameter second is always non-negative after implicit conversion